connection_provider_pgx.go

```diff
@@ -101,6 +96,11 @@ func (p *PgxConnectionProvider) Close() {
 	p.pools = make(map[string]*pgxpool.Pool)
 }
 
+// GetNoRowsSentinel implements ConnectionProvider.GetNoRowsSentinel.
+func (*PgxConnectionProvider) GetNoRowsSentinel() error {
+	return pgx.ErrNoRows
+}
```

Comment:
As stated elsewhere: I suspect that if you do `errors.Is(err, sql.ErrNoRows)` with a pgx `err`, it would work. Which means you wouldn't need this function. See the code here: https://github.com/jackc/pgx/blob/61d3c965ad442cc14d6b0e39e0ab3821f3684c03/conn.go#L106-L125

That being said, I also understand your decisions to not rely on that, since pgx does it, but others might not.

Reply:
Yup, I'd actually prefer to keep this driver-agnostic 🙂 we allow our end users to define manual connection providers and having the power to say what "no rows found error" is is actually very empowering.